src/Core/NationalCloud.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ */
+
+namespace Microsoft\Graph\Core;
+
+/**
+ * Class NationalCloud
+ *
+ * Defines Graph hosts for the various national clouds
+ *
+ * @package Microsoft\Graph\Core
+ * @copyright 2021 Microsoft Corporation
+ * @license https://opensource.org/licenses/MIT MIT License
+ * @link https://developer.microsoft.com/graph
+ */
+final class NationalCloud
+{
+    const GLOBAL = "https://graph.microsoft.com";
+    const US_GOV = "https://graph.microsoft.us";
+    const US_DOD = "https://dod-graph.microsoft.us";
+    const GERMANY = "https://graph.microsoft.de";
+    const CHINA = "https://microsoftgraph.chinacloudapi.cn";
+
+    /**
+     * Stores all enum values as list
+     * Prevents having to do reflection for each call to getValues()
+     *
+     * @var array
+     */
+    private static $values = [];
+
+    /**
+     * Returns a list of the constant values
+     *
+     * @return array
+     */
+    public static function getValues(): array {
```

Comment:
Do you think it is a good idea to have a boolean function returning if the url contains a valid national cloud host? 
`IsValidNationalHost` 
This can be moved to that function ->
```
if (!in_array($nationalCloud, NationalCloud::getValues())) {
            throw new ClientInitialisationException("Invalid national cloud passed. See NationalCloud constants");
```
In case you need to reuse the same logic in multiple places.

Reply:
Why not store the values as a collection directly? What is the need to populate the list everytime? 

Also please consider the following comment I received suggesting use of set : 
https://github.com/microsoftgraph/msgraph-sdk-javascript/pull/336#discussion_r505150958
I don't know if this is efficient in PHP

Reply:
I like the idea of a boolean function.

Reply:
"Why not store the values as a collection directly? What is the need to populate the list everytime?"

By this do you mean initialise the values list with the constants e.g.
```php
$values = [
    NationalCloud::GLOBAL,
    NationalCoud::US_GOV,
    ...
]
```

Reply:
There is no built-in Set object in PHP. Do you still think it is worth the investment to implement this?
I would do so if the input was somewhat dynamic and we needed to guarantee unique items.

Reply:
"What is the need to populate the list everytime?"

We populate it once the first time `getValues()` is called since reflection is potentially expensive

Reply:
There is no built in Set object in PHP, however I can use a dictionary and lookup the keys instead
```php
$values = [
    NationalCloud::GLOBAL => true,
    NationalCloud::US_GOV => true,
    ...
]
```

Reply:
> "Why not store the values as a collection directly?"
> 
> By this do you mean initialise the values list with the constants? e.g.
> 
> ```
> $values = [
>     NationalCloud::GLOBAL,
>     NationalCoud::US_GOV,
>     ...
> ]
> ```

Yes.

Reply:
This would not be possible.
I need to do some processing to each constant's url before adding it to the list hence using a method to initialise this only once.